openmc/mgxs/mdgxs.py

```diff
@@ -0,0 +1,1564 @@
+        if 'energy low [MeV]' in df and 'energyout low [MeV]' in df:
+            df.rename(columns={'energy low [MeV]': 'group in'},
+                      inplace=True)
+            in_groups = np.tile(all_groups, self.num_subdomains * num_delayed_groups)
+            in_groups = np.repeat(in_groups, df.shape[0] / in_groups.size)
+            df['group in'] = in_groups
+            del df['energy high [MeV]']
+
+            df.rename(columns={'energyout low [MeV]': 'group out'},
+                      inplace=True)
+            out_groups = np.repeat(all_groups, self.xs_tally.num_scores)
+            out_groups = np.tile(out_groups, df.shape[0] / out_groups.size * num_delayed_groups)
+            df['group out'] = out_groups
+            del df['energyout high [MeV]']
+            columns = ['group in', 'group out']
+
+        elif 'energyout low [MeV]' in df:
+            df.rename(columns={'energyout low [MeV]': 'group out'},
+                      inplace=True)
+            in_groups = np.tile(all_groups, self.num_subdomains * num_delayed_groups)
+            df['group out'] = in_groups
+            del df['energyout high [MeV]']
+            columns = ['group out']
+
+        elif 'energy low [MeV]' in df:
+            df.rename(columns={'energy low [MeV]': 'group in'}, inplace=True)
+            in_groups = np.tile(all_groups, self.num_subdomains * num_delayed_groups)
+            df['group in'] = in_groups
+            del df['energy high [MeV]']
+            columns = ['group in']
+
+        # Select out those groups the user requested
+        if not isinstance(groups, basestring):
+            if 'group in' in df:
+                df = df[df['group in'].isin(groups)]
+            if 'group out' in df:
+                df = df[df['group out'].isin(groups)]
+
+        # If user requested micro cross sections, divide out the atom densities
+        if xs_type == 'micro':
+            if self.by_nuclide:
+                densities = self.get_nuclide_densities(nuclides)
+            else:
+                densities = self.get_nuclide_densities('sum')
+            densities = np.repeat(densities, len(self.rxn_rate_tally.scores))
+            tile_factor = df.shape[0] / len(densities)
+            df['mean'] /= np.tile(densities, tile_factor)
+            df['std. dev.'] /= np.tile(densities, tile_factor)
+
+        # Sort the dataframe by domain type id (e.g., distribcell id) and
+        # energy groups such that data is from fast to thermal
+        if self.domain_type == 'mesh':
+            mesh_str = 'mesh {0}'.format(self.domain.id)
+            df.sort_values(by=[(mesh_str, 'x'), (mesh_str, 'y'), \
+                               (mesh_str, 'z')] + columns, inplace=True)
+        else:
+            df.sort_values(by=[self.domain_type] + columns, inplace=True)
+        return df
+
+        # Select out those delayed groups the user requested
+        if not isinstance(delayed_groups, basestring):
```

Comment:
Much of the code is redundant, but some of the operations require the `num_delayed_groups` for tiling the column values for the energy groups. I'll look into seeing if I can call the `MGXS` class `get_pandas_dataframe(...)` method and then just re-tile the column values.